optuna/samplers/_motpe.py

```diff
@@ -0,0 +1,627 @@
+        seed:
+            Seed for random number generator.
+
+    .. note::
+        Initialization with Latin hypercube sampling may improve optimization performance.
+        However, the current implementation only supports initialization with random sampling.
+
+    Example:
+
+        .. testcode::
+
+            import optuna
+
+            seed = 128
+            num_variables = 9
+            n_startup_trials = 11 * num_variables - 1
+
+
+            def objective(trial):
+                x = []
+                for i in range(1, num_variables + 1):
+                    x.append(trial.suggest_float(f"x{i}", 0.0, 2.0 * i))
+                return x
+
+
+            sampler = optuna.samplers.MOTPESampler(
+                n_startup_trials=n_startup_trials, n_ehvi_candidates=24, seed=seed
+            )
+            study = optuna.create_study(directions=["minimize"] * num_variables, sampler=sampler)
+            study.optimize(objective, n_trials=250)
+    """
+
+    def __init__(
+        self,
+        *,
+        consider_prior: bool = True,
+        prior_weight: float = 1.0,
+        consider_magic_clip: bool = True,
+        consider_endpoints: bool = True,
+        n_startup_trials: int = 10,
+        n_ehvi_candidates: int = 24,
+        gamma: Callable[[int], int] = default_gamma,
+        weights_above: Callable[[int], np.ndarray] = _default_weights_above,
+        seed: Optional[int] = None,
+    ) -> None:
+
+        super().__init__(
+            consider_prior=consider_prior,
+            prior_weight=prior_weight,
+            consider_magic_clip=consider_magic_clip,
+            consider_endpoints=consider_endpoints,
+            n_startup_trials=n_startup_trials,
+            n_ei_candidates=n_ehvi_candidates,
+            gamma=gamma,
+            weights=weights_above,
+            seed=seed,
+        )
+        self._n_ehvi_candidates = n_ehvi_candidates
+        self._mo_random_sampler = RandomSampler(seed=seed)
+
+    def reseed_rng(self) -> None:
+        self._rng = np.random.RandomState()
+        self._mo_random_sampler.reseed_rng()
+
+    def infer_relative_search_space(
+        self,
+        study: optuna.study.Study,
+        trial: optuna.trial.FrozenTrial,
+    ) -> Dict[str, BaseDistribution]:
+        return {}
+
+    def sample_relative(
+        self,
+        study: optuna.study.Study,
+        trial: optuna.trial.FrozenTrial,
+        search_space: Dict[str, BaseDistribution],
+    ) -> Dict[str, Any]:
+        return {}
+
+    def sample_independent(
+        self,
+        study: optuna.study.Study,
+        trial: optuna.trial.FrozenTrial,
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+        if len(study.directions) < 2:
+            raise ValueError(
+                "Number of objectives must be >= 2. "
+                "Please use optuna.samplers.TPESampler for single-objective optimization."
+            ) from None
```

Comment:
Not introduced by this PR so a minor suggestion. I don't think we're in any error handling context so there's no exception to suppress?
```suggestion
            )
```